sympy/functions/elementary/hyperbolic.py

```diff
@@ -761,6 +761,9 @@ def taylor_term(n, x, *previous_terms):
     def _eval_rewrite_as_sinh(self, arg):
         return S.ImaginaryUnit / sinh(arg + S.ImaginaryUnit * S.Pi /2)
 
+    def _eval_is_finite(self):
+        return self.args[0].is_real
```

Comment:
This is wrong, as it would return `False` for everything except `reals`.

```python
In [3]: x = symbols('x', real=False)

In [4]: sech(x).is_finite
Out[4]: False
```
Also:
```python
In [11]: sech(5 + I).is_finite
Out[11]: False
```
But sech(5 + I) is finite:
```
In [12]: sech(5 + I).n()
Out[12]: 0.00728166229072838 - 0.011339487429296*I
```